src/lib/apexPmd.ts

```diff
@@ -128,24 +128,30 @@ export class ApexPmd {
     const rulesetsArg = this.rulesets.join(',');
 
     const cacheKey = enableCache ? `--cache "${cachePath}"` : '--no-cache';
+    const noProgressBar = '--no-progress';
     const formatKey = `-f csv`;
     const targetPathKey = `-d "${targetPath}"`;
     const rulesetsKey = `-R "${rulesetsArg}"`;
 
-    const pmdKeys = `${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
+    const pmdKeys = `${noProgressBar} ${formatKey} ${cacheKey} ${targetPathKey} ${rulesetsKey}`;
 
     const classPath = [
-      path.join(pmdBinPath, 'lib', '*'),
       path.join(workspaceRootPath, '*'),
       ...additionalClassPaths,
     ].join(CLASSPATH_DELM);
 
-    const javaExc = this.config.jrePath ? path.join(this.config.jrePath, 'bin', 'java') : 'java';
-    const cmd = `${javaExc} -cp "${classPath}" net.sourceforge.pmd.PMD ${pmdKeys}`;
+    let env : NodeJS.ProcessEnv = {};
+    env["CLASSPATH"] = classPath;
+    if (this.config.jrePath) {
+      env["JAVA_HOME"] = this.config.jrePath;
```

Comment:
we have to actually set the `JAVA_HOME` & `CLASSPATH` env directly?

Reply:
I'm using now the script bundled with PMD to start PMD (https://github.com/pmd/pmd/blob/master/pmd-dist/src/main/resources/scripts/pmd). Setting the env `CLASSPATH` is the way add additional libraries onto the classpath (like custom rules). PMD's own classpath (e.g. the files in lib/*) is handled by the script.

But looking at `JAVA_HOME` - it doesn't seem to be necessary, as we just run java in the end: https://github.com/pmd/pmd/blob/0557c5c85eccdb98577590e3425170e236646649/pmd-dist/src/main/resources/scripts/pmd#L212

But that means, that setting `JAVA_HOME` has no effect. We would need to set `PATH=$JAVA_HOME/bin` instead. It seems, we introduced JAVA_HOME in 2008 but really never used it.

I'll try to rework that part tomorrow.

Reply:
I've updated the code to now actually set PATH instead with the proper JAVA_HOME/bin. Now configuring a special JRE (using preference `apexPMD.jrePath`) is working again.